airflow/plugins_manager.py

```diff
@@ -164,8 +163,28 @@ def load_plugins_from_plugin_directory():
     global plugins  # pylint: disable=global-statement
     log.debug("Loading plugins from directory: %s", settings.PLUGINS_FOLDER)
 
+    patterns_by_dir: Dict[str, List[Pattern[str]]] = {}
+
     # Crawl through the plugins folder to find AirflowPlugin derivatives
-    for root, _, files in os.walk(settings.PLUGINS_FOLDER, followlinks=True):  # noqa # pylint: disable=too-many-nested-blocks
+    for root, dirs, files in os.walk(settings.PLUGINS_FOLDER, followlinks=True):  # noqa # pylint: disable=too-many-nested-blocks
+
+        patterns: List[Pattern[str]] = patterns_by_dir.get(root, [])
+        ignore_file = os.path.join(root, '.pluginignore')
+
+        if os.path.isfile(ignore_file):
+            with open(ignore_file, 'r') as file:
```

Comment:
Hello.
This code looks very similar to:
https://github.com/apache/airflow/blob/master/airflow/utils/file.py#L125
I think you can make a generator that will take the starting path and yield for each file to load. 
Best regards,
Kamil

Reply:
You are right.
In fact, this is pretty much the same code.

 I make a generator. What do you think of this? 
（Perhaps we can make the generator func share with file.py but....）

Reply:
I like this code now. :-)

Moving this method to airflow.utils.file is a good idea. This will allow us to delete the duplicate code.